projects/igniteui-angular/src/lib/core/styles/components/progress/linear/_linear-theme.scss

```diff
@@ -0,0 +1,506 @@
+        fill-color-danger: $fill-color-danger,
+        fill-color-warning: $fill-color-warning,
+        fill-color-info: $fill-color-info,
+        fill-color-success: $fill-color-success,
+        stripes-color: $stripes-color,
+        text-color: $text-color,
+        track-border-radius: $track-border-radius,
+        theme: map.get($schema, '_meta', 'theme'),
+        _meta: map.merge(if($meta, $meta, ()), (
+            variant: map.get($schema, '_meta', 'theme')
+        )),
+    ));
+}
+
+@mixin striped-gradient($variant: default, $gradient-orientation, $stripe-color) {
+    $fill-color-default: if($variant == 'indigo', transparent, $stripe-color);
+    $stripes-color: if($variant == 'indigo', $stripe-color, transparent);
+
+    & {
+        background-image: repeating-linear-gradient(
+            $gradient-orientation,
+            $fill-color-default 0,
+            $fill-color-default var(--stripe-size),
+            $stripes-color var(--stripe-size),
+            $stripes-color calc(var(--stripe-size) * 2)
+        );
+    }
+}
+
+$easing-curves: (
+    // Primary translate easing curves
+    primary-translate-start: cubic-bezier(0.5, 0, 0.7017, 0.4958),
+    primary-translate-mid: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563),
+
+    // Primary scale easing curves
+    primary-scale-slow-start: cubic-bezier(0.3347, 0.124, 0.7858, 1),
+    primary-scale-quick-end: cubic-bezier(0.06, 0.11, 0.6, 1),
+
+    // Secondary translate easing curves
+    secondary-translate-start: cubic-bezier(0.15, 0, 0.515, 0.4096),
+    secondary-translate-mid: cubic-bezier(0.31, 0.284, 0.8, 0.7337),
+    secondary-translate-end: cubic-bezier(0.4, 0.627, 0.6, 0.902),
+
+    // Secondary scale easing curves
+    secondary-scale-slow-start: cubic-bezier(0.15, 0, 0.515, 0.4096),
+    secondary-scale-mid: cubic-bezier(0.31, 0.284, 0.8, 0.7337),
+    secondary-scale-smooth-end: cubic-bezier(0.4, 0.627, 0.6, 0.902)
+);
+
+// Helper function to retrieve easing curves
+@function get-easing($curve) {
+    @if not map.has-key($easing-curves, $curve) {
+        @warn 'Easing curve #{$curve} does not exist.';
+    }
+    @return map.get($easing-curves, $curve);
+}
+
+/// @deprecated Use the `css-vars` mixin instead.
+/// @see {mixin} css-vars
+/// @param {Map} $theme - The theme used to style the component.
+@mixin progress-linear($theme) {
+    @include css-vars($theme);
+
+    $variant: map.get($theme, '_meta', 'variant');
+
+    $bar-height: map.get((
+        'material': rem(4px),
+        'fluent': rem(2px),
+        'bootstrap': rem(16px),
+        'indigo': rem(4px),
+    ), $variant);
+
+    $bar-height: map.get((
+        'material': rem(4px),
+        'fluent': rem(2px),
+        'bootstrap': rem(16px),
+        'indigo': rem(4px),
+    ), $variant);
+
+    $strip-size: map.get((
+        'material': rem(16px),
+        'fluent': rem(16px),
+        'bootstrap': rem(5px),
+        'indigo': rem(16px),
+    ), $variant);
```

Comment:
Ideally, all of these should be listed as design tokens in the component schema. Let's defer this for now and add it to another task for when we refactor the token system in the theming library.

Reply:
I am adding a TODO so we don't forget about this